### PR TITLE
kvrpc: allow batch lock resolve

### PR DESCRIPTION
**Pull request by atmzhou**
The new Protocol allows batch lock resolve.

### CODE REVIEW

**Comment by atmzhou** [comment]
@zhangjinpeng1987 @wentaoxu

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -176,11 +176,18 @@ message ScanLockResponse {
...
+    uint64 txn = 1;
+    uint64 status = 2;
+}
+
 message ResolveLockRequest {
     Context context = 1;
     uint64 start_version  = 2;
     // If the txn is rolled back, do not set it.
     uint64 commit_version = 3;
+    uint64 single_or_batch = 4;
```
Comment (disksing) [inline proto/kvrpcpb.proto]: I don't think we need this field. Check if `start_version` is 0 should be enough.

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -176,11 +176,17 @@ message ScanLockResponse {
     repeated LockInfo locks = 3;
 }
 
+message Txn2Status {
```
Comment (zhangjinpeng1987) [inline proto/kvrpcpb.proto]: s/Txn2Status/TxnStatus

Comment (atmzhou) [reply proto/kvrpcpb.proto]: TxnStatus has been used by tidb as an alias for uint64.
So we must use another name.

File: proto/kvrpcpb.proto [hunk 3]
```diff
@@ -176,11 +176,17 @@ message ScanLockResponse {
...
+    uint64 txn = 1;
+    uint64 status = 2;
+}
+
 message ResolveLockRequest {
     Context context = 1;
     uint64 start_version  = 2;
     // If the txn is rolled back, do not set it.
     uint64 commit_version = 3;
+    repeated Txn2Status txn2statusS = 4;
```
Comment (zhangjinpeng1987) [inline proto/kvrpcpb.proto]: s/txn2statusS/multi_txn_status/

File: proto/kvrpcpb.proto (hunk 2 above)
Comment (siddontang) [reply proto/kvrpcpb.proto]: use a better name, not 2.

Comment (atmzhou) [reply proto/kvrpcpb.proto]: how about TxnInfo?
@siddontang

**Comment by atmzhou** [comment]
@zhangjinpeng1987 @wentaoxu

**Comment by disksing** [comment]
LGTM.

File: proto/kvrpcpb.proto [hunk 4]
```diff
@@ -176,11 +176,17 @@ message ScanLockResponse {
...
+    uint64 txn = 1;
+    uint64 status = 2;
+}
+
 message ResolveLockRequest {
     Context context = 1;
     uint64 start_version  = 2;
     // If the txn is rolled back, do not set it.
     uint64 commit_version = 3;
+    repeated TxnInfo multi_txn_status = 4;
```
Comment (zhangjinpeng1987) [inline proto/kvrpcpb.proto]: s/multi_txn_status/txn_infos

**Comment by atmzhou** [comment]
@zhangjinpeng1987

**Pull request merged**